systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,333 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+                        new GenericKafkaListenerBuilder()
+                            .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                            .withPort(9094)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(true)
+                            .withAuth(new KafkaListenerAuthenticationTls())
+                            .build())
+                .endKafka()
+            .endSpec()
+            .build());
+
+        // Deploy target kafka with tls listeners with tls and SCRAM-SHA authentication
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(kafkaClusterTargetName, 3, 3)
+            .editSpec()
+                .editKafka()
+                    .withListeners(
+                        new GenericKafkaListenerBuilder()
+                            .withName(kafkaTlsScramListener)
+                            .withPort(9093)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(true)
+                            .withAuth(new KafkaListenerAuthenticationScramSha512())
+                            .build(),
+                        new GenericKafkaListenerBuilder()
+                            .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                            .withPort(9094)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(true)
+                            .withAuth(new KafkaListenerAuthenticationTls())
+                            .build())
+                .endKafka()
+            .endSpec()
+            .build());
+
+        // Deploy topic
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(kafkaClusterSourceName, topicSourceName, 3).build());
+
+        // Create Kafka user for source cluster
+        KafkaUser userSource = KafkaUserTemplates.scramShaUser(kafkaClusterSourceName, kafkaUserSource).build();
+        resourceManager.createResource(extensionContext, userSource);
+        KafkaUser userTlsSource = KafkaUserTemplates.tlsUser(kafkaClusterSourceName, kafkaUserTlsSource).build();
+        resourceManager.createResource(extensionContext, userTlsSource);
+
+        // Create Kafka user for target cluster
+        KafkaUser userTarget = KafkaUserTemplates.scramShaUser(kafkaClusterTargetName, kafkaUserTarget).build();
+        resourceManager.createResource(extensionContext, userTarget);
+        KafkaUser userTlsTarget = KafkaUserTemplates.tlsUser(kafkaClusterTargetName, kafkaUserTlsTarget).build();
+        resourceManager.createResource(extensionContext, userTlsTarget);
+
+        // Initialize PasswordSecretSource to set this as PasswordSecret in MirrorMaker2 spec
+        PasswordSecretSource passwordSecretSource = new PasswordSecretSource();
+        passwordSecretSource.setSecretName(kafkaUserSource);
+        passwordSecretSource.setPassword("password");
```

Comment:
Would it make sense to add some variable for the `password`?

Reply:
Might create it, as it's not the first usage in STs..

Reply:
I mean some test variable, you don't have to create a constant

Reply:
Naah, not that important at my use-case.